service-seed-without-router/all-actors/src/main/java/org/sunbird/BaseActor.java

```diff
@@ -1,43 +1,55 @@
 package org.sunbird;
 
 import akka.actor.UntypedAbstractActor;
-import org.apache.log4j.Logger;
+import akka.event.DiagnosticLoggingAdapter;
+import akka.event.Logging;
 import org.sunbird.message.IResponseMessage;
 import org.sunbird.message.Localizer;
 import org.sunbird.message.ResponseCode;
 import org.sunbird.request.Request;
 
+import java.util.HashMap;
 import java.util.Locale;
+import java.util.Map;
+import java.util.UUID;
 
 /**
  * @author Amit Kumar
  */
 public abstract class BaseActor extends UntypedAbstractActor {
 
-    private Logger logger = Logger.getLogger(BaseActor.class);
+    final DiagnosticLoggingAdapter logger = Logging.getLogger(this);
+
     public abstract void onReceive(Request request) throws Throwable;
+
     protected Localizer localizer = Localizer.getInstance();
 
     @Override
     public void onReceive(Object message) throws Throwable {
+        Map<String, Object> mdc = new HashMap<>();
+        mdc.put("requestId", UUID.randomUUID().toString());
+        logger.setMDC(mdc);
         if (message instanceof Request) {
             Request request = (Request) message;
             String operation = request.getOperation();
-            logger.info("BaseActor:onReceive called for operation:" + operation);
             try {
-                logger.info(String.format("%s:%s:method started at %s",this.getClass().getSimpleName(),operation,System.currentTimeMillis()));
+                logger.info("Started : operation {}", operation);
                 onReceive(request);
-                logger.info(String.format("%s:%s:method ended at %s",this.getClass().getSimpleName(),operation,System.currentTimeMillis()));
+                logger.info("Ended : operation {}", operation);
             } catch (Exception e) {
+                logger.debug("Exception : operation {} {}", operation, e.getMessage());
```

Comment:
this must be error - and just get entire exception printed, not just its message.